client/src/cmdhfseos.c

```diff
@@ -163,7 +163,10 @@ static int decrypt_cryptogram(uint8_t *key, uint8_t *input, uint8_t *out, int in
 }
 
 static void increment_command_wrapper(uint8_t *input, int input_len) {
-    input[input_len - 1]++; // Increment the last element of the header by 1
+    // Increment the end of the header by 1
```

Comment:
I could also split the `&input[input_len - sizeof(uint32_t)]` out onto a separate line, if you'd like. Maybe something like:
```c
    uint8_t* input_end = &input[input_len - sizeof(uint32_t)];
    uint32_t value = MemBeToUint4byte(input_end);
    value++;
    Uint4byteToMemBe(input_end, value);
```

Reply:
yeah ,  that will make it even more readable.  if you rename it from `input_end` to `offset`   I merge :)